homeassistant/components/isy994/switch.py

```diff
@@ -87,3 +120,62 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     def icon(self) -> str:
         """Get the icon for programs."""
         return "mdi:script-text-outline"  # Matches isy program icon
+
+
+class ISYEnableSwitchEntity(ISYAuxControlEntity, SwitchEntity):
+    """A representation of an ISY program switch."""
+
+    def __init__(
+        self,
+        node: Node,
+        control: str,
+        unique_id: str,
+        description: EntityDescription,
+        device_info: DeviceInfo | None,
+    ) -> None:
+        """Initialize the ISY Aux Control Number entity."""
+        super().__init__(
+            node=node,
+            control=control,
+            unique_id=unique_id,
+            description=description,
+            device_info=device_info,
+        )
+        self._attr_name = description.name  # Override super
+        self._change_handler: EventListener = None
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to the node control change events."""
+        self._change_handler = self._node.isy.nodes.status_events.subscribe(
+            self.async_on_update,
+            event_filter={
+                TAG_ADDRESS: self._node.address,
+                ATTR_ACTION: NC_NODE_ENABLED,
+            },
+            key=self.unique_id,
+        )
+
+    @callback
+    def async_on_update(self, event: NodeChangedEvent, key: str) -> None:
+        """Handle a control event from the ISY Node."""
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        return True  # Enable switch is always available
+
+    @property
+    def is_on(self) -> bool | None:
+        """Get whether the ISY device is in the on state."""
+        return bool(self._node.enabled)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Send the turn off command to the ISY switch."""
+        if not await self._node.disable():
+            _LOGGER.debug("Unable to disable device")
```

Comment:
https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=silver#silver-

Operations like service calls and entity methods (e.g. Set HVAC Mode) have proper exception handling. Raise ValueError on invalid user input and raise HomeAssistantError for other failures such as a problem communicating with a device.